index.rst

```diff
@@ -1,61 +1,201 @@
-..
-  Technote content.
+:tocdepth: 1
 
-  See https://developer.lsst.io/restructuredtext/style.html
-  for a guide to reStructuredText writing.
+.. sectnum::
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+Abstract
+========
 
-  Use the following syntax for sections:
+Given the authentication implementation described in DMTN-169_, Butler then has to decide whether to permit an operation on a collection following requirement DMS-REQ-0340 in LSE-61_.
+This document proposes having the Butler service maintain a list of groups and do a group membership check of the requesting user.
+Other options and their trade-offs are also discussed.
 
-  Sections
-  ========
+.. _DMTN-169: https://dmtn-169.lsst.io/
+.. _LSE-61: https://docushare.lsst.org/docushare/dsweb/Get/LSE-61
 
-  and
+Requirements
+============
 
-  Subsections
-  -----------
+DMS-REQ-0340 states:
 
-  and
+    All Level 3 data products shall be configured to have the ability to have access restricted to the owner, a list of people, a named group, or be completely public.
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+Per LPM-231_, "Level 3" here refers to User Generated data products.
+The Butler service will also serve Data Release data products.
+Those will be public and read-only.
 
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+.. _LPM-231: https://docushare.lsst.org/docushare/dsweb/Get/LPM-231
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+A local Butler instance does not impose any access control and instead relies on the underlying file system permissions.
+However, the Butler service as described in DMTN-169_ will receive authenticated requests from users and must decide whether to allow an operation while meeting the DMS-REQ-0340 requirement.
 
-     Caption text.
+The Butler organizes data into collections, which support read and write operations.
+All Butler operations are done on collections.
```

Comment:
Actually, dataset types may be even simpler than that, because they can probably always be world-readable, so we may just be able to use the user/group namespacing rules with no ACLs.

Reply:
We only care about data products from a DMS-REQ-0340 perspective, so my inclination is to change this sentence to instead say "All Butler data operations are done on collections." It doesn't sound like there's much security concern about access to dataset types or dimension rows (correct me if I'm wrong), so I think you can handle them however you want.

If you want the ACL policy for them to be recorded in this document as well so that everything's in the same place, could you propose what you want to say? If I'm understanding the problem correctly, I think saying whatever you would prefer to say is fine.

Reply:
Sounds good.  Might be best for @timj and I to just create a new ticket or two, for one of us to add those details to this document once we've thought them through, and ask you to make sure nothing raises any flags.

Reply:
Okay, I'll change this to data operations for now, and then am happy to review those additions.